crates/rspack_core/src/compiler/compilation.rs

```diff
@@ -445,11 +445,15 @@ impl Compilation {
           .module_graph
           .dependency_by_id(&id)
           .expect("dependency not found");
+        if dependency.as_module_dependency().is_none()
+          && dependency.as_context_dependency().is_none()
```

Comment:
Could we also impl `trait ModuleDependency` to context dependencies ?

Reply:
why would you think so? All context dependencies in webpack extend from ﻿`ContextDependency`.

Reply:
Actually in webpack not only ModuleDependency can resolve a module, Dependency which implement getResourceIdentifier() and return a non-null value can resolve a module

Reply:
> Actually in webpack not only ModuleDependency can resolve a module, Dependency which implement getResourceIdentifier() and return a non-null value can resolve a module

I know this and kinda of weird. Just wondering why context dependencies not using Module dependency abstraction.